src/compute/src/render/mod.rs

```diff
@@ -616,6 +589,77 @@ where
     }
 }
 
+use differential_dataflow::dynamic::pointstamp::PointStamp;
+
+impl<G> Context<G, Row>
+where
+    G: Scope<Timestamp = Product<mz_repr::Timestamp, PointStamp<usize>>>,
+{
+    /// Renders a plan to a differential dataflow, producing the collection of results.
+    ///
+    /// This method allows for `plan` to contain a `LetRec` variant at its root, and is planned
+    /// in the context of `level` pre-existing iteration coordinates.
+    ///
+    /// This method recursively descends `LetRec` nodes, establishing nested scopes for each
+    /// and establishing the appropriate recursive dependencies among the bound variables.
+    /// Once non-`LetRec` nodes are reached it calls in to `render_plan` which will error if
+    /// furher `LetRec` variants are found.
```

Comment:
I see, this probably answers my comment below! Is this something that we cannot support, or is it something that can never happen?

Reply:
It is in principle supportable with some more code complexity (teaching `render_plan` about different timestamp types), but not currently a thing that we need, as we can always push logic wrapping a `LetRec` down into the operator by having it wrap `body` instead, with I think identical information communicated about the plan. If we learn we would prefer not to do that for any reason we can revisit.